pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -65,119 +76,205 @@ func NewIngressAnalyzerWithObjects(objects []scan.K8sObject, pe *eval.PolicyEngi
 /////////////////////////////////////////////////////////////////////////////////////////////////
 // services analysis
 
-const (
-	selectorError        = "selector conversion error"
-	missingSelectorError = "error : K8s Service without selectors is not supported"
-)
+const missingSelectorWarning = "K8s Service without selectors is not supported"
 
-// this function populates servicesToPeersMap
+// mapServiceToPeers populates servicesToPeersMap
 func (ia *IngressAnalyzer) mapServiceToPeers(svc *corev1.Service) error {
 	// get peers selected by the service selectors
 	peers, err := ia.getServicePeers(svc)
 	if err != nil {
 		return err
 	}
+	if len(peers) == 0 {
+		// service was ignored
+		return nil
+	}
 	if _, ok := ia.servicesToPeersMap[svc.Namespace]; !ok {
 		ia.servicesToPeersMap[svc.Namespace] = make(map[string][]eval.Peer)
 	}
 	ia.servicesToPeersMap[svc.Namespace][svc.Name] = peers
 	return nil
 }
 
+// getServicePeers given a service return its selected peers
 func (ia *IngressAnalyzer) getServicePeers(svc *corev1.Service) ([]eval.Peer, error) {
 	svcStr := types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}.String()
 	if svc.Spec.Selector == nil {
-		return nil, errors.New(scan.Service + " " + svcStr + " " + missingSelectorError)
+		ia.logger.Warnf("ignoring " + scan.Service + " " + svcStr + ": " + missingSelectorWarning)
```

Comment:
this warning was initially an error - so we had a test -"TestNotSupportedService" , since the warning is raised by the logger now, I didn't check the warning message but that the servicesMap is empty.
Since we don't warn if there are no policies/ policy-rules selecting pod I didn't warn if a service doesn't actually select any peer / if route/Ingress don't select any service (warning raised only if the target resource is not supported , i.e. not a service)

Reply:
> this warning was initially an error - so we had a test -"TestNotSupportedService" , since the warning is raised by the logger now, I didn't check the warning message (testing logger messages is a little bit complicated and I was afraid to have a time waste) but that the servicesMap is empty. Since we don't warn if there are no policies/ policy-rules selecting pod I didn't warn if a service doesn't actually select any peer / if route/Ingress don't select any service (warning raised only if the target resource is not supported , i.e. not a service)